mapper/merge_list_by_index.go

```diff
@@ -0,0 +1,111 @@
+package mapper
+
+import (
+	"fmt"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+	"github.com/rancher/norman/types/definition"
+	"github.com/rancher/norman/types/mapper"
+)
+
+func NewMergeListByIndexMapper(From, To string, Ignores ...string) *MergeListByIndexMapper {
+	rtn := MergeListByIndexMapper{
+		From:       From,
+		To:         To,
+		Ignore:     make(map[string]struct{}),
+		fromFields: []string{},
+	}
+	for _, Ignore := range Ignores {
+		rtn.Ignore[Ignore] = struct{}{}
+	}
+	return &rtn
+}
+
+type MergeListByIndexMapper struct {
+	From       string
+	To         string
+	Ignore     map[string]struct{}
+	fromFields []string
+}
+
+func (m *MergeListByIndexMapper) FromInternal(data map[string]interface{}) {
+	fromObj, ok := data[m.From]
+	if !ok {
+		return
+	}
+	toObj, ok := data[m.To]
+	if !ok {
+		return
+	}
+	fromList := convert.ToMapSlice(fromObj)
+	toList := convert.ToMapSlice(toObj)
+	for i := 0; i < len(fromList) && i < len(toList); i++ {
+		fromItem := fromList[i]
+		toItem := toList[i]
```

Comment:
If I understand correctly this mapper is for merging metric spec & status to the same entry. How do we guarantee that this `fromList[i]` and `toList[i]` is referencing the same metric?

Reply:
It is from kubernetes code. https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/horizontal.go#L240